module/VuFind/src/VuFind/Controller/AbstractSearch.php

```diff
@@ -252,6 +244,13 @@ protected function getSearchSetupCallback()
                     $config[$key] = $value;
                 }
             }
+
+            // Special case: override recommend settings through parameter (used by
+            // combined search)
+            if ($override) {
```

Comment:
Maybe `if (is_array($override)) {` to prevent unexpected behavior if invalid parameters are passed in?

Reply:
I made this change.  I'm a little skeptical because that check wasn't done in the legacy code so it changes the behavior, and I'm not sure $override being an array tells us too much about it having valid content.  But it's still incremental improvement, so probably no harm.

Reply:
My goal is to avoid fatal errors. If you give the code a junk array, the junk data will be ignored. But if you pass a non-array to array_merge, you'll blow everything up entirely. :-)